node_modules/oae-principals/lib/api.user.js

```diff
@@ -130,52 +133,70 @@ var createUser = module.exports.createUser = function(ctx, tenantAlias, displayN
     validator.check(opts.visibility, {'code': 400, 'msg': 'The specified visibility setting is unknown'}).isIn(_.values(AuthzConstants.visibility));
     validator.check(opts.emailPreference, {'code': 400, 'msg': 'The specified email preference is invalid'}).isIn(_.values(PrincipalsConstants.emailPreferences));
 
+    // Because some SSO strategies do not release an email address, we allow user accounts to be
+    // created without providing the email. The UI however, will keep nagging users about filling
+    // one in
+    if (opts.email) {
+        validator.check(opts.email, {'code': 400, 'msg': 'The specified email address is invalid'}).isEmail();
+    }
     if (validator.hasErrors()) {
         return callback(validator.getFirstError());
     }
 
-    var id = AuthzUtil.toId('u', tenantAlias, ShortId.generate());
-
-    var values = {
-        'tenantAlias': tenantAlias,
-        'displayName': displayName,
-        'visibility': opts.visibility,
-        'email': opts.email,
-        'emailPreference': opts.emailPreference,
-        'locale': opts.locale,
-        'publicAlias': opts.publicAlias,
-        'smallPictureUri': opts.smallPictureUri,
-        'mediumPictureUri': opts.mediumPictureUri,
-        'largePictureUri': opts.lagePictureUri
-    };
+    // Verify this email address, if any, is not already mapped to another user
```

Comment:
Can you just drop a `TODO` in here for this nagging to happen. It's hard to predict how much time will be in between this and a fully featured nag workflow.